src/Disks/IO/AsynchronousReadIndirectBufferFromRemoteFS.cpp

```diff
@@ -164,6 +164,8 @@ bool AsynchronousReadIndirectBufferFromRemoteFS::nextImpl()
     if (!hasPendingDataToRead())
         return false;
 
+    assert(file_offset_of_buffer_end <= impl->getFileSize());
```

Comment:
As discussed, we can add below assert under an if that there are no tasks in threadpool
assert(file_offset_of_buffer_end >= impl->getImplementationBufferOffset());